brayns/io/ColorMapLoader.cpp

```diff
@@ -0,0 +1,121 @@
+/* Copyright (c) 2015-2016, EPFL/Blue Brain Project
+ * All rights reserved. Do not distribute without permission.
+ * Responsible Author: Cyrille Favreau <cyrille.favreau@epfl.ch>
+ *
+ * This file is part of Brayns <https://github.com/BlueBrain/Brayns>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include <brayns/common/log.h>
+#include <brayns/common/material/Texture2D.h>
+#include <brayns/common/scene/Scene.h>
+
+#include <fstream>
+
+#include "ColorMapLoader.h"
+
+namespace brayns
+{
+
+const uint DEFAULT_ALPHA = 255;
+
+ColorMapLoader::ColorMapLoader()
+{
+}
+
+const Vector4uis& ColorMapLoader::getValues() const
+{
+    return _values;
+}
+
+void ColorMapLoader::setValues( const Vector4uis& values )
+{
+    _values.clear();
+    for( auto value: values )
+        _values.push_back( value );
+}
+
+bool ColorMapLoader::loadFromFile( const std::string& filename )
+{
+    BRAYNS_INFO << "Loading color map from " << filename << std::endl;
+    std::ifstream file( filename, std::ios::in );
+    if( !file.good( ))
+    {
+        BRAYNS_ERROR << "Could not open file " << filename << std::endl;
+        return false;
+    }
+
+    bool validParsing = true;
+    std::string line;
+
+    while( validParsing && std::getline( file, line ))
+    {
+        std::vector< uint > lineData;
+        std::stringstream lineStream(line);
+
+        size_t value;
+        while( lineStream >> value )
+            lineData.push_back(value);
+
+        switch( lineData.size() )
+        {
+        case 3:
+            _values.push_back( Vector4ui(
+                lineData[0], lineData[1], lineData[2], DEFAULT_ALPHA ));
+            break;
+        case 4:
+            _values.push_back( Vector4ui(
+                lineData[0], lineData[1], lineData[2], lineData[3] ));
+            break;
+        default:
+            BRAYNS_ERROR << "Invalid line: " << line << std::endl;
+            validParsing = false;
+            break;
+        }
+    }
+
+    file.close();
+    return validParsing;
+}
```

Comment:
There's something I don't understand here. After the discussion the other day I thought we had agreed on trying to take the use cases related to colormaps as an attempt to have one real shared component (and here I'm referring to a full library solution, not just an event schema). When we talked the other day about the colormap in Brayns, I thought you just wanted to do a small experiment with hardcoded values to understand better how it could work in OSPRay, but now you want to merge in 300 lines of code with the full feature. I know you want to try things, but I don't understand why you want to merge into master our third implementation of the same thing instead of making the effort in converging into a common solution. With this approach, the convergence won't ever happen. Was any user pressing to get simulation mapping in Brayns?